mesh/mesh.cpp

```diff
@@ -3595,11 +3595,20 @@ Mesh::Mesh(Mesh *orig_mesh, int ref_factor, int ref_type)
       }
    }
 
-   SetCurvature(1, true, spaceDim);
-   Vector node_coordinates_vec(
-      node_coordinates.Data(),
-      node_coordinates.Width()*node_coordinates.Height());
-   SetNodes(node_coordinates_vec);
+   if (orig_mesh->GetNodes())
+   {
+      Vector node_coordinates_vec(
+         node_coordinates.Data(),
+         node_coordinates.Width()*node_coordinates.Height());
+      L2_FECollection fec_dg(1, Dim, BasisType::GaussLobatto);
+      FiniteElementSpace fes_dg(this, &fec_dg, spaceDim, 1);
+      GridFunction nodes_dg(&fes_dg);
+      nodes_dg = node_coordinates_vec;
```

Comment:
You can avoid the copy using:
```c++
   GridFunction nodes_dg;
   nodes_dg.MakeRef(&fes_dg, node_coordinates.Data());
```
This also avoids the need for `node_coordinates_vec`.

Reply:
Good suggestion — this both avoids the copy and simplifies the code. Thanks.